### PR TITLE
Add the Reek checker for Ruby.

### PR DESCRIPTION
**Pull request by elebow**
Add support for Reek, a Ruby code smell detector.

### CODE REVIEW

File: ale_linters/ruby/reek.vim [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        endfor
+    endfor
+
+    return l:output
+endfunction
+
+function! s:BuildText(error) abort
+    let l:text = a:error.smell_type . ':'
+
+    if get(g:, 'ale_ruby_reek_show_context', 0)
```
Comment (w0rp) [inline ale_linters/ruby/reek.vim]: You should set some default values for these at the top of the file, like in the other files, and then use `ale#Var` here. `ale#Var` is being used so all settings can be overridden in each buffer.

File: ale_linters/ruby/reek.vim [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    endif
+
+    let l:errors = json_decode(a:lines[0])
+
+    let l:output = []
+
+    for l:error in l:errors
+        for l:location in l:error.lines
+            call add(l:output, {
+            \    'bufnr': a:buffer,
```
Comment (w0rp) [inline ale_linters/ruby/reek.vim]: You don't have to set `bufnr` if you don't want to. That key will be set automatically by ALE later on.

File: test/handler/test_reek_handler.vader [hunk 3]
```diff
@@ -0,0 +1,67 @@
+Execute(The reek handler should parse JSON correctly):
+    runtime ale_linters/ruby/reek.vim
+
+    let g:ale_ruby_reek_show_context = 1
+    let g:ale_ruby_reek_show_wiki_link = 0
```
Comment (w0rp) [inline test/handler/test_reek_handler.vader]: You should split each of these checks into separate `Execute` blocks. That way, the tests can fail later for 1 out of 4 cases, and we can spot which one is broken.

**Comment by elebow** [comment]
That should address the feedback. Do you want me to squash the fixup commits?

**Comment by w0rp** [comment]
Nah, I'd rather have a bunch of commits in a pull request. Then I check check out "changes since your last review" in GitHub. I tend to squash merge pull requests.

**Comment by w0rp** [comment]
Cheers! 🍻

**Pull request merged**